File: typescript/flatten-array/flatten-array.ts
export default class FlattenArray {
    static flatten(arr: any): number[] {
        return arr.reduce((acc: any, e:any): number[] => {
            if (Array.isArray(e)) return [...acc, ...this.flatten(e)]
            if (typeof e === "number") acc.push(e)
            return acc
        }, [])
    }
}

File: go/flatten-array/flatten_array.go
package flatten

func Flatten(array interface{}) []interface{}  {
	flattened := []interface{}{}
	for _, v := range array.([]interface{})	 {
		switch vv := v.(type) {
		case []interface{}:
			flattened = append(flattened, Flatten(vv)...)
		case int:
			flattened = append(flattened, vv)
		}
	}

	return flattened
}